runtime/recipe-index.js

```diff
@@ -12,38 +12,79 @@ import {Loader} from './loader.js';
 import {Manifest} from './manifest.js';
 import {Arc} from './arc.js';
 import {SlotComposer} from './slot-composer.js';
-import {Strategizer} from '../strategizer/strategizer.js';
+import {Strategizer, Strategy} from '../strategizer/strategizer.js';
 import {StrategyExplorerAdapter} from './debug/strategy-explorer-adapter.js';
 import {Tracing} from '../tracelib/trace.js';
 import {ConvertConstraintsToConnections} from './strategies/convert-constraints-to-connections.js';
 import {ResolveRecipe} from './strategies/resolve-recipe.js';
-import {InitPopulation} from './strategies/init-population.js';
+import {CreateHandleGroup} from './strategies/create-handle-group.js';
 import {AddUseHandles} from './strategies/add-use-handles.js';
 import * as Rulesets from './strategies/rulesets.js';
 import {DevtoolsConnection} from './debug/devtools-connection.js';
 import {RecipeUtil} from './recipe/recipe-util.js';
 import {Handle} from './recipe/handle.js';
 
+class RelevantContextRecipes extends Strategy {
+  constructor(context, affordance) {
+    super();
+    this._recipes = [];
+    for (let recipe of context.recipes) {
+      if (recipe.particles.find(p => p.spec && !p.spec.matchAffordance(affordance)) !== undefined) {
+        continue;
+      }
+
+      recipe = recipe.clone();
+      let options = {errors: new Map()};
+      if (!recipe.normalize(options)) {
```

Comment:
nit (sorry) - positive condition is preferable

Reply:
Done :)